st2api/st2api/controllers/resource.py

```diff
@@ -187,6 +198,16 @@ def _get_by_name(self, resource_name):
 
         return resource_db
 
+    def _get_from_model_kwargs_for_request(self, request):
+        """
+        Retrieve kwargs which are passed to "LiveActionAPI.model" method.
+
+        :param request: Pecan request object.
+
+        :rtype: ``dict``
+        """
+        return self.from_model_kwargs
```

Comment:
This is a base implementation so it doesn't necessary need to do anything with the `request` argument.

First I went with the approach you have described in the executions controller, but later on, I moved `from_model_kwargs` from class attribute to inline so it's a bit more efficient.